### PR TITLE
Provide support for *.hjson

### PR DESCRIPTION
**Pull request by winston0410**
Everything in master branch and everything work this time.

### CODE REVIEW

File: queries/hjson/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,16 @@
+(true) @boolean
+(false) @boolean
+(null) @constant.builtin
+(number) @number
+(pair key: (string) @label)
+(pair value: (string) @string)
+(array (string) @string)
+;  (string_content (escape_sequence) @string.escape)
```
Comment (stsewd) [inline queries/hjson/highlights.scm:8]: are these not used or do you plan to support them?

**Review by stsewd — COMMENTED** [review]
I think you are missing adding the parser itself (parsers.lua)

**Comment by winston0410** [comment]
Right I forgot to commit `parser.lua`, added back in now.

File: queries/hjson/highlights.scm (hunk 1 above)
Comment (winston0410) [reply queries/hjson/highlights.scm:8]: I am still figuring out how to handle quoteless string and the escape sequence, I do plan to support them and as everything other than that is usable now, I have just commented it out there.

**Comment by vigoux** [comment]
Could you also add a `ftdetect` for this filetype ? We now tend to provide some kind of out-of-the-bow support.

**Comment by winston0410** [comment]
Typo there. Added back in the right vim file now.

**Review by vigoux — APPROVED** [review]

**Comment by winston0410** [comment]
It seems like the error from CI is not related with my commit?

**Comment by vigoux** [comment]
It's strange that the error about jsonc impacts you...

**Comment by vigoux** [comment]
Could you try rebasing on master just to see ?

**Comment by winston0410** [comment]
It seems like I cannot rebase it, as my changes are already starting from the latest commit from upstream master.

**Comment by stsewd** [comment]
I had the same problem in https://github.com/nvim-treesitter/nvim-treesitter/pull/1651, but I'm not sure how to fix it. I think maybe the npm install step is breaking things somehow...

**Comment by theHamsta** [comment]
@winston0410 friendly ping. I think this is an awesome addition. Could you resolve the conflicts?

File: queries/hjson/indents.scm [hunk 2]
```diff
@@ -0,0 +1,3 @@
+; inherits: json
+
+(comment) @ignore
```
Comment (theHamsta) [inline queries/hjson/indents.scm:3]: Ignore?

Comment (stsewd) [reply queries/hjson/indents.scm:3]: This is so comments aren't taken into consideration to calculate the current indentation level, in particular when the comment is a multiline comment.

**Comment by winston0410** [comment]
@theHamsta sure. Do you want me to do this in a new PR, because I forget the create a new branch for this one, I guess it cannot rebase properly.

**Comment by theHamsta** [comment]
What ever is easier for you. You can also just do a `git reset --soft origin/master` on this branch and apply and commit. For rebase and reset you always have to force push. You can also resolve the conflicts here in the online editor

**Comment by winston0410** [comment]
Closing this in flavour of https://github.com/nvim-treesitter/nvim-treesitter/pull/1836

**Pull request closed without merging**